esp-wifi/src/compat/misc.rs

```diff
@@ -20,8 +21,26 @@ unsafe extern "C" fn fclose(stream: *const ()) -> i32 {
     todo!("fclose {:?}", stream);
 }
 
-// not available in ROM on ESP32-S2
-#[cfg(feature = "esp32s2")]
+// We cannot just use the ROM function since it needs to allocate memory
+#[no_mangle]
+unsafe extern "C" fn strdup(str: *const i8) -> *const u8 {
+    trace!("strdup {:?}", str);
+
+    unsafe {
+        let s = core::ffi::CStr::from_ptr(str);
+        let s = s.to_str().unwrap();
```

Comment:
Any reason not to use `count_bytes` here instead? It'd be ideal not to assume UTF-8 here by calling `to_str()`

Reply:
Agreed - not only are C strings not necessarily UTF8, the validation done by `to_str` is annoyingly expensive.

Reply:
yeah - I just re-surrected this from an old commit without looking at it